tests/unit/events/test_event_listener.py

```diff
@@ -185,6 +186,28 @@ def test_context_manager_multiple(self):
 
         assert EventBus.event_listeners == [e1]
 
+    def test_threaded(self):
```

Comment:
This is great!

Could you also add a test for not using `copy_context_vars`?
And can you add the same with/without tests, but for creating a thread directly instead of a pool?
(And if possible, could you do the same with/without tests for coroutines?)

I know this is kind of neurotic, but I want tests for each of these situations so that we double check we don't have any invalid assumptions.